### PR TITLE
Simplify checks like "x is not None or y is not None or z is not None"

### PR DESCRIPTION
**Pull request by michaelgrund**
**Description of proposed changes**

Related to #2757 

- [x] text.py

Fixes #

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Comment by michaelgrund** [comment]
Do we want to have this simplifications only for 3 checks like 
```
x is not None or y is not None or z is not None:
```
or also for 2 checks like:
```
x is not None or y is not None:
```
?

**Comment by seisman** [comment]
> also for 2 checks like:
> 
> ```
> x is not None or y is not None:
> ```

I feel it's not necessary to rewrite 2-checks.

**Review by yvonnefroehlich — COMMENTED** [review]
Probably also the lines 194-197 can be rewritten using `any`:
https://github.com/GenericMappingTools/pygmt/blob/579ad70c17d11541922d1c728202c518e3d2682d/pygmt/src/text.py#L192-L200

(I am sorry in case my post at https://github.com/GenericMappingTools/pygmt/issues/2757#issuecomment-1793690935 was misleading. It was meant only as a question whether I understood this issue correctly, not as a complete update of the `Figure.text` method.)

File: pygmt/src/text.py [hunk 1]
```diff
@@ -190,11 +190,8 @@ def text_(
...
 
     # Build the -F option in gmt text.
-    if kwargs.get("F") is None and (
-        position is not None
-        or angle is not None
-        or font is not None
-        or justify is not None
+    if kwargs.get("F") is None and 
+        any(v is not None for v in (position, angle, font, justify))
     ):
```
Comment (yvonnefroehlich) [inline pygmt/src/text.py:195]:
```suggestion
    if kwargs.get("F") is None and any(
        v is not None for v in (position, angle, font, justify)
    ):
```

File: pygmt/src/text.py [hunk 2]
```diff
@@ -189,11 +189,8 @@ def text_(  # noqa: PLR0912
...
 
     # Build the -F option in gmt text.
-    if kwargs.get("F") is None and (
-        position is not None
-        or angle is not None
-        or font is not None
-        or justify is not None
+    if kwargs.get("F") is None and 
+        any(v is not None for v in (position, angle, font, justify)
     ):
```
Comment (yvonnefroehlich) [inline pygmt/src/text.py:194]:
```suggestion
    if kwargs.get("F") is None and any(
        v is not None for v in (position, angle, font, justify)
    ):
```

**Review by seisman — APPROVED** [review]

**Pull request merged**